### PR TITLE
Fix grade upgrade fail announcement

### PR DESCRIPTION
**Pull request by eppc0330**
* **Addressed Issue(s)**: 

* **Server Mode**: Renewal

* **Description of Pull Request**: 

    Now grade upgrade fail message will be shown only when upgrading to grade A and B.
    This is official behavior.

### CODE REVIEW

**Comment by secretdataz** [comment]
Shouldn't this be done in enchantgrade.yml instead?

**Comment by eppc0330** [comment]
> Shouldn't this be done in enchantgrade.yml instead?

This is not affecting success message. Only fail message.

**Comment by Atemo** [comment]
In that case how about replacing `Announce` field by `AnnounceSuccess` and `AnnounceFail` in the yml file to make things clearer?

**Comment by eppc0330** [comment]
> In that case how about replacing `Announce` field by `AnnounceSuccess` and `AnnounceFail` in the yml file to make things clearer?

Right this seems good

**Comment by eppc0330** [comment]
Sorry miss click

File: src/map/status.cpp [hunk 1]
```diff
@@ -674,17 +674,31 @@ uint64 EnchantgradeDatabase::parseBodyNode( const ryml::NodeRef& node ){
 				}
 			}
 
-			if( this->nodeExists( gradeNode, "Announce" ) ){
-				bool announce;
+			if( this->nodeExists( gradeNode, "AnnounceSuccess" ) ){
+				bool announcesucess;
```
Comment (aleos89) [inline src/map/status.cpp:678]:
```suggestion
				bool announcesuccess;
```

File: src/map/status.cpp [hunk 2]
```diff
@@ -674,17 +674,31 @@ uint64 EnchantgradeDatabase::parseBodyNode( const ryml::NodeRef& node ){
...
 
-				if( !this->asBool( gradeNode, "Announce", announce ) ){
+				if( !this->asBool( gradeNode, "AnnounceSuccess", announcesucess ) ){
```
Comment (aleos89) [inline src/map/status.cpp:680]:
```suggestion
				if( !this->asBool( gradeNode, "AnnounceSuccess", announcesuccess ) ){
```

**Review by aleos89 — APPROVED** [review]

**Review by Atemo — APPROVED** [review]

**Comment by Lemongrass3110** [comment]
Needs some small adjustments. I will push them to your branch this weekend!

File: db/enchantgrade.yml [hunk 3]
```diff
@@ -30,7 +30,9 @@
 #           Refine                      Required refine level.
 #           Chance                      Base chance of success out of 0~10000.
 #           Bonus                       Enchantgrade bonus. (Default: 0)
-#           Announce                    Announce if someone tries to increase the enchantgrade. (Default: true)
+#           AnnounceSuccess             Announce on upgrade success. (Default: true)
+#           AnnounceFail                Announce on upgrade failure. (Default: false)
+#           Announce                    Announce on upgrade success and failure.
```
Comment (Atemo) [inline db/enchantgrade.yml:35]: Seems a bit redundant as the 2 above do the job, but why not.
The rest looks good !

**Review by Atemo — APPROVED** [review]

File: db/enchantgrade.yml (hunk 3 above)
Comment (Lemongrass3110) [reply db/enchantgrade.yml:35]: Thats just to keep support for V1. :)

**Comment by eppc0330** [comment]
Sorry I deleted wrong repository... I can't edit this pull anymore T_T

**Pull request merged**